avm/ptn/network/hub-networking/modules/subnets.bicep

```diff
@@ -0,0 +1,178 @@
+    ipAllocations: ipAllocations
+    serviceEndpointPolicies: serviceEndpointPolicies
+  }
+}
+
+resource subnet_roleAssignments 'Microsoft.Authorization/roleAssignments@2022-04-01' = [
+  for (roleAssignment, index) in (formattedRoleAssignments ?? []): {
+    name: roleAssignment.?name ?? guid(subnet.id, roleAssignment.principalId, roleAssignment.roleDefinitionId)
+    properties: {
+      roleDefinitionId: roleAssignment.roleDefinitionId
+      principalId: roleAssignment.principalId
+      description: roleAssignment.?description
+      principalType: roleAssignment.?principalType
+      condition: roleAssignment.?condition
+      conditionVersion: !empty(roleAssignment.?condition) ? (roleAssignment.?conditionVersion ?? '2.0') : null // Must only be set if condtion is set
+      delegatedManagedIdentityResourceId: roleAssignment.?delegatedManagedIdentityResourceId
+    }
+    scope: subnet
+  }
+]
+
+@description('The resource group the subnet was deployed into.')
+output resourceGroupName string = resourceGroup().name
+
+@description('The name of the subnet.')
+output subnetName string = subnet.name
+
+@description('The resource ID of the subnet.')
+output resourceId string = subnet.id
+
+@description('The address prefix for the subnet.')
+output subnetAddressPrefix string = subnet.properties.addressPrefix
+
+@description('List of address prefixes for the subnet.')
+output subnetAddressPrefixes array = !empty(addressPrefixes) ? subnet.properties.addressPrefixes : []
+
+// =============== //
+//   Definitions   //
+// =============== //
+
+type roleAssignmentType = {
+  @description('Optional. The name (as GUID) of the role assignment. If not provided, a GUID will be generated.')
+  name: string?
+  @description('Required. The role to assign. You can provide either the display name of the role definition, the role definition GUID, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'.')
+  roleDefinitionIdOrName: string
+  @description('Required. The principal ID of the principal (user/group/identity) to assign the role to.')
+  principalId: string
+  @description('Optional. The principal type of the assigned principal ID.')
+  principalType: ('ServicePrincipal' | 'Group' | 'User' | 'ForeignGroup' | 'Device')?
+  @description('Optional. The description of the role assignment.')
+  description: string?
+  @description('Optional. The conditions on the role assignment. This limits the resources it can be assigned to. e.g.: @Resource[Microsoft.Storage/storageAccounts/blobServices/containers:ContainerName] StringEqualsIgnoreCase "foo_storage_container".')
+  condition: string?
+  @description('Optional. Version of the condition.')
+  conditionVersion: '2.0'?
+  @description('Optional. The Resource Id of the delegated managed identity resource.')
+  delegatedManagedIdentityResourceId: string?
```

Comment:
```suggestion
  @description('Optional. The name (as GUID) of the role assignment. If not provided, a GUID will be generated.')
  name: string?
  
  @description('Required. The role to assign. You can provide either the display name of the role definition, the role definition GUID, or its fully qualified ID in the following format: \'/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11\'.')
  roleDefinitionIdOrName: string
  
  @description('Required. The principal ID of the principal (user/group/identity) to assign the role to.')
  principalId: string
  
  @description('Optional. The principal type of the assigned principal ID.')
  principalType: ('ServicePrincipal' | 'Group' | 'User' | 'ForeignGroup' | 'Device')?
  
  @description('Optional. The description of the role assignment.')
  description: string?
  
  @description('Optional. The conditions on the role assignment. This limits the resources it can be assigned to. e.g.: @Resource[Microsoft.Storage/storageAccounts/blobServices/containers:ContainerName] StringEqualsIgnoreCase "foo_storage_container".')
  condition: string?
  
  @description('Optional. Version of the condition.')
  conditionVersion: '2.0'?
  
  @description('Optional. The Resource Id of the delegated managed identity resource.')
  delegatedManagedIdentityResourceId: string?
```

Reply:
Fixed.